src/us/kbase/workspace/test/database/provenance/ContributorTest.java

```diff
@@ -0,0 +1,741 @@
+			"\t",
+			null,
+			"Credit:FundingAcquisition"
+	);
+
+	// invalid contributor role strings
+	static final List<String> INVALID_ROLES = Arrays.asList(
+		"magical fairy princess",
+		"ContributorRole:FUNDING_ACQUISITION",
+		"credit:",
+		"datecite:software",
+		"credit:workpackageleader"
+	);
+
+	static final List<String> INVALID_ROLE_ERRORS = INVALID_ROLES.stream().map(r -> "Invalid contributorRole: " + r)
+			.collect(Collectors.toList());
+
+
+	static final List<String> INVALID_ROLES_WITH_WS_NULL = Arrays.asList(
+				"magical fairy princess",
+			"    ",
+			"ContributorRole:FUNDING_ACQUISITION",
+				"credit:",
+			"\n\n\n",
+			"datecite:software",
+				"",
+			null,
+			"credit:workpackageleader",
+			"datecite:software"
+	);
+
+	static final Map<String, String> NAME_MAP = ImmutableMap.of(NAME, NAME_STRING);
+
+	static final Map<String, String> NAME_ID_MAP = ImmutableMap.of(NAME, NAME_STRING, CONTRIB_ID,
+			CONTRIB_ID_STRING);
+
+	static final Map<String, String> NAME_CREDIT_MAP = ImmutableMap.of(
+			NAME, NAME_STRING,
+			CREDIT_NAME, CREDIT_NAME_STRING);
+
+	static final Map<String, String> ALL_MAP = ImmutableMap.of(
+			CREDIT_NAME, CREDIT_NAME_STRING,
+			NAME, NAME_STRING,
+			CONTRIB_ID, CONTRIB_ID_STRING);
+
+	@Test
+	public void equals() throws Exception {
+		EqualsVerifier.forClass(Contributor.class).usingGetClass().verify();
+	}
+
+	/**
+	 * Check the contributor has the expected fields
+	 *
+	 * @param contributor
+	 *                instance to check
+	 * @param ct
+	 *                expected contributor type
+	 * @param expectedMap
+	 *                key/value pairs are the field names and values for the
+	 *                string fields that are expected not to be null
+	 * @param orgs
+	 *                list of Organization objects expected to be returned by
+	 *                `getAffiliations`;
+	 *                if null, `getAffiliations` should return an empty list
+	 * @param roles
+	 *                list of ContributorRoles expected to be returned by
+	 *                `getContributorRoles`;
+	 *                if null, `getContributorRoles` should return an empty list
+	 */
+	private void assertContributorFields(final Contributor contributor, final ContributorType ct,
+			final Map<String, String> expectedMap, final List<Organization> orgs,
+			final List<ContributorRole> roles, final List<String> roleStrings) {
+		assertThat(INCORRECT_CONTRIB_TYPE, contributor.getContributorType(), is(ct));
+		assertThat(INCORRECT_NAME, contributor.getName(), is(expectedMap.get(NAME)));
+		assertThat(INCORRECT_CREDIT_NAME, contributor.getCreditName(),
+				is(expectedMap.containsKey(CREDIT_NAME) ? opt(expectedMap.get(CREDIT_NAME)) : ES));
+		assertThat(INCORRECT_CONTRIB_ID, contributor.getContributorID(),
+				is(expectedMap.containsKey(CONTRIB_ID) ? opt(expectedMap.get(CONTRIB_ID)) : ES));
+
+		assertThat(INCORRECT_AFFILIATIONS, contributor.getAffiliations(),
+				is(orgs == null ? Collections.emptyList() : orgs));
```

Comment:
the typing has been done now, so it's a moot point

Reply:
Sorry, wasn't clear - the issue to me is that logical constructs should be avoided in tests to avoid recapitulating the code under test. Here, it'd be better if the arguments to the test helper were either orgs or an empty list rather than null.

For the 2nd part I mean that you can avoid typing Collections.emptyList() over and over in the tests that call the helper by adding a class var EL = Collections.emptyList()

Reply:
ah, got it. It's probably better to be explicit and verbose in PR comments as I didn't realise that was what you were talking about.

Reply:
:+1: